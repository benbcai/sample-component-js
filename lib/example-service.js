class ExampleService {
  before() {
    console.log('*********Executing "before" hook in Example Service*********');
  }
}

module.exports = ExampleService;
