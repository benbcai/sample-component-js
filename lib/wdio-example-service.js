class CustomService {
  before(capabilities) {
    console.log('*********Executing "before" hook in Custom Service*********');
  }
}

module.exports = CustomService;
