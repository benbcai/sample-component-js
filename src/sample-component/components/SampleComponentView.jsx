import React from 'react';
import PropTypes from 'prop-types';
import { ApplicationIntlContext } from 'orion-application/lib/application-intl';
import ContentContainer from 'terra-content-container';
import Header from 'terra-action-header';
import Spacer from 'terra-spacer';
import StatusView from 'terra-status-view';

const propTypes = {
  /**
   * Example data.
   */
  exampleData: PropTypes.string,
};

const SampleComponentView = ({ exampleData }) => {
  const intl = React.useContext(ApplicationIntlContext);
  const headerTitle = intl.formatMessage({ id: 'sample-component-engine.sample-component' });

  return (
    <ContentContainer header={<Header title={headerTitle} />}>
      <Spacer paddingTop="large+1" paddingBottom="large+1" paddingLeft="large+1" paddingRight="large+1">
        {exampleData}
        {!exampleData && (
          <StatusView
            variant="no-data"
            isGlyphHidden
            message={intl.formatMessage({ id: 'sample-component-engine.no-sample-component' })}
          />
        )}
      </Spacer>
    </ContentContainer>
  );
};

SampleComponentView.propTypes = propTypes;

export default SampleComponentView;
