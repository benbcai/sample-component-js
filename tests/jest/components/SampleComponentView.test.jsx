import React from 'react';
import { mockIntl } from 'terra-enzyme-intl';
import { ApplicationIntlContext } from 'orion-application/lib/application-intl';
import SampleComponentView from '../../../src/sample-component/components/SampleComponentView';

it('should render a SampleComponentView with no data', () => {
  jest.spyOn(React, 'useContext').mockImplementation((contextType) => {
    if (contextType === ApplicationIntlContext) {
      return mockIntl;
    }
    return undefined;
  });
  const sampleComponentView = shallow(<SampleComponentView />);
  expect(sampleComponentView.first().shallow()).toMatchSnapshot();
});

it('should render a SampleComponentView that has succeeded with a valid SampleComponentEngine', () => {
  jest.spyOn(React, 'useContext').mockImplementation((contextType) => {
    if (contextType === ApplicationIntlContext) {
      return mockIntl;
    }
    return undefined;
  });
  const sampleComponentView = shallow(<SampleComponentView exampleData="SampleComponent" />);
  expect(sampleComponentView.first().shallow()).toMatchSnapshot();
});
