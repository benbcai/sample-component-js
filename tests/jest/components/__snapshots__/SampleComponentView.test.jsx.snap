// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a SampleComponentView that has succeeded with a valid SampleComponentEngine 1`] = `
<div
  className="content-container-static"
>
  <div
    className="header"
  >
    <ActionHeader
      level={1}
      title="sample-component-engine.sample-component"
    />
  </div>
  <div
    className="main"
  >
    <Scroll
      className="normalizer"
    >
      <Spacer
        isInlineBlock={false}
        marginBottom="none"
        marginLeft="none"
        marginRight="none"
        marginTop="none"
        paddingBottom="large+1"
        paddingLeft="large+1"
        paddingRight="large+1"
        paddingTop="large+1"
      >
        SampleComponent
      </Spacer>
    </Scroll>
  </div>
</div>
`;

exports[`should render a SampleComponentView with no data 1`] = `
<div
  className="content-container-static"
>
  <div
    className="header"
  >
    <ActionHeader
      level={1}
      title="sample-component-engine.sample-component"
    />
  </div>
  <div
    className="main"
  >
    <Scroll
      className="normalizer"
    >
      <Spacer
        isInlineBlock={false}
        marginBottom="none"
        marginLeft="none"
        marginRight="none"
        marginTop="none"
        paddingBottom="large+1"
        paddingLeft="large+1"
        paddingRight="large+1"
        paddingTop="large+1"
      >
        <StatusView
          buttonAttrs={Array []}
          isAlignedTop={false}
          isGlyphHidden={true}
          message="sample-component-engine.no-sample-component"
          variant="no-data"
        />
      </Spacer>
    </Scroll>
  </div>
</div>
`;
